src/explore-education-statistics-admin/src/pages/release/content/components/ReleaseContent.tsx

```diff
@@ -26,23 +26,51 @@ import Details from '@common/components/Details';
 import PageSearchForm from '@common/components/PageSearchForm';
 import RelatedAside from '@common/components/RelatedAside';
 import ReleaseDataAndFilesAccordion from '@common/modules/release/components/ReleaseDataAndFilesAccordion';
-import React, { useCallback, useMemo } from 'react';
-import { generatePath, useLocation } from 'react-router';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+import { generatePath, Prompt, useLocation } from 'react-router';
 
 interface MethodologyLink {
   key: string;
   title: string;
   url: string;
 }
 
+const handleBeforeUnload = (showPrompt: boolean) => {
+  window.onbeforeunload = event => {
+    if (showPrompt) {
+      event.preventDefault();
+      // eslint-disable-next-line no-param-reassign
+      event.returnValue = '';
+    }
+  };
+};
+
 const ReleaseContent = () => {
   const config = useConfig();
   const location = useLocation();
-
-  const { editingMode } = useEditingContext();
+  const {
+    editingMode,
+    unsavedBlocks,
+    unsavedCommentDeletions,
+  } = useEditingContext();
   const { release } = useReleaseContentState();
   const actions = useReleaseContentActions();
 
+  const [showPrompt, setShowPrompt] = useState<boolean>(false);
+
+  useEffect(() => {
+    const blocksWithCommentDeletions = Object.entries(unsavedCommentDeletions)
+      .filter(blockWithDeletions => blockWithDeletions[1].length)
+      .map(blockWithDeletions => blockWithDeletions[0]);
+    setShowPrompt(
+      unsavedBlocks.length > 0 || blocksWithCommentDeletions.length > 0,
+    );
+  }, [setShowPrompt, unsavedBlocks, unsavedCommentDeletions]);
+
+  useEffect(() => {
+    handleBeforeUnload(showPrompt);
```

Comment:
Seems like we're missing the cleanup callback to remove the event handler from the window when this component unmounts.

Reply:
Done